### DISCUSSION TITLE
Brownie does not run Ganache

### DISCUSSION THREAD

**Discussion by taigaisatiger** [smartcontractkit/full-blockchain-solidity-course-py#1316] [Q&A]
On lesson 5 on 4:36:32, when I run the code below,

`brownie run scripts/deploy.py`

it does not run ganache, and it does not spit out an address and private key. I've managed to overcome errors up until this point running snippets of code that are online, but I cannot find an answer to this one. And also, it may have been affected by the snippets of code that I pasted.

```
taiga@taigaMacBook-Pro brownie_simple_storage % brownie run scripts/deploy.py
Brownie v1.16.4 - Python development framework for Ethereum

BrownieSimpleStorageProject is the active project.
Attached to local RPC client listening at '127.0.0.1:8545'...

Running 'scripts/deploy.py::main'...
```

I am not really sure of the reason behind this phenomenon. Could someone please help me with this problem?

**Comment by cromewar** [comment]
Hello @taigaisatiger 
Is ganache-cli added to your path?  please run `ganache-cli` and see if it throws any errors.
Also the message says `127.0.0.1:8545` does the script works if you have ganache ui running?

**Reply by taigaisatiger** [reply]
Hello thank you for your reply.

`zsh: command not found: ganache-cli`

This is the response I get when I run `ganache-cli`.

I was able to use ganache previously by doing `npm install -D` and then `npx ganache-cli`. This is where I got this solution from (https://stackoverflow.com/questions/69921649/command-not-found-ganache-cli).

When I run `npm install -g ganache-cli`, or when I run `npm install -g ganache` I get this response below.

```
taiga@taigaMacBook-Pro brownie_simple_storage % npm install -g ganache-cli
npm ERR! code EEXIST
npm ERR! path /Users/taiga/.npm-global/bin/ganache-cli
npm ERR! EEXIST: file already exists
npm ERR! File exists: /Users/taiga/.npm-global/bin/ganache-cli
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/taiga/.npm/_logs/2022-03-21T14_04_29_778Z-debug-0.log
```

So this is my situation right now.

`Also the message says 127.0.0.1:8545 does the script works if you have ganache ui running?`
→ I am not sure how I am supposed to check this or whether I am answering your question, but the desktop version is not running because I ⌘Q the desktop version.

**Reply by cromewar** [reply]
Yes the problem is with your ganache, as brownie could not find it on your system.
So Please take this info in consideration:
1. Is better to install node with nvm or brew.
2. Install ganache globally with `npm i -g ganache-cli`.
3. Be sure your node modules are added correctly to your path.

I highly recommend nvm: https://github.com/nvm-sh/nvm

**Reply by taigaisatiger** [reply]
For 1. when I run `node  --verison` it gives me `v16.14.2`.
For 2. when I run `npm i -g ganache-cli`, I get 

```
npm WARN deprecated ganache-cli@6.12.2: ganache-cli is now ganache; visit https://trfl.io/g7 for details

changed 6 packages, and audited 102 packages in 2s

2 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (7 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```
So is there a problem with npm?
When I run `npm --version` it gives me `8.5.0`, and there is no problem, so I am having a hard time identifying the problem.

I've deleted the ganache-cli multiple times and ran `npm i -g ganache-cli`, but it gives me the same response everytime.

**Reply by cromewar** [reply]
Yes I think the problem is with your node installation, I strongly suggest to delete your node installation and install it over again using NVM, use the link I posted above as a guide.

**Reply by taigaisatiger** [reply]
my nvm version is 0.39.1

**Reply by taigaisatiger** [reply]
I still couldn't work this out.

**Reply by taigaisatiger** [reply]
So much errors

**Reply by taigaisatiger** [reply]
I ran this `npm uninstall ganache-cli -g && npm install ganache -g`, but it is an error again.
What am I supposed to do?
It's taking me days just to do this. Not learning anything but just passing time.
So frustrating....



```
up to date, audited 1 package in 648ms

found 0 vulnerabilities
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /Users/taiga/.npm-global/lib/node_modules/ganache
npm ERR! dest /Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u
npm ERR! errno -66
npm ERR! ENOTEMPTY: directory not empty, rename '/Users/taiga/.npm-global/lib/node_modules/ganache' -> '/Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u'

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/taiga/.npm/_logs/2022-03-22T02_39_41_844Z-debug-0.log
```

**Reply by cromewar** [reply]
There is not a package ganache just ganache-cli so the command you posted above was too large.

**Reply by taigaisatiger** [reply]
If you look at this document it says ganache-cli turned into ganache

https://github.com/trufflesuite/ganache/releases/tag/v7.0.0

**Reply by taigaisatiger** [reply]
I ran this code `npm uninstall ganache-cli -g && npm install ganache-cli -g` but same results:(





```
up to date, audited 1 package in 648ms

found 0 vulnerabilities
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /Users/taiga/.npm-global/lib/node_modules/ganache
npm ERR! dest /Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u
npm ERR! errno -66
npm ERR! ENOTEMPTY: directory not empty, rename '/Users/taiga/.npm-global/lib/node_modules/ganache' -> '/Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u'

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/taiga/.npm/_logs/2022-03-22T02_39_41_844Z-debug-0.log
taiga@taiganoMacBook-Pro brownie_simple_storage % npm uninstall ganache-cli -g && npm install ganache-cli -g

up to date, audited 1 package in 679ms

found 0 vulnerabilities
npm WARN deprecated ganache-cli@6.12.2: ganache-cli is now ganache; visit https://trfl.io/g7 for details

added 1 package, and audited 102 packages in 4s

2 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (7 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

**Reply by taigaisatiger** [reply]
It says ganache-cli is deprecated as well on nmp

https://www.npmjs.com/package/ganache-cli

**Reply by cromewar** [reply]
Use this command on your terminal and then try again:
`rm -rf /Users/taiga/.npm-global/lib/node_modules/`
this should empty you npm `node_modules` folder and get rid of your conflicts.

**Reply by taigaisatiger** [reply]
I deleted it, ran `npm install ganache --global` , but now it spits out `zsh: command not found: ganache`.
Does this mean the installation process fail?
I see ganache inside  `node_modules` so I am confused whether it succeeded or not. Do you know?

**Reply by cromewar** [reply]
As you have .zsh and not .bash did you add nvm to path as nvm suggest?
![image](https://user-images.githubusercontent.com/20199180/159510154-4ace79e8-421b-47d2-917f-d352a0179edf.png)
